src/redux/reducers/extensions.js

```diff
@@ -1,18 +1,17 @@
-const defaultState = {};
+const defaultState = {}
 
 const extensions = (state = defaultState, action) => {
   switch (action.type) {
     case 'SET_EXTENSION_DATA':
-      const extensionName = action.extension;
-      const currentData = state[extensionName] || {};
-      const updatedData = Object.assign({}, currentData, action.data);
-      const updatedState = state;
-      state[extensionName] = updatedData;
-
-      return Object.assign({}, state, updatedState);
+      const extensionName = action.extension
+      const currentData = state[extensionName] || {}
+      const updatedData = Object.assign({}, currentData, action.data)
+      return Object.assign({}, state, {
+        extensionName: updatedData,
```

Comment:
`extensionName` in the object will have a key of `extensionName` instead of `action.extension`'s value.

I think you want something like:

```
let updatedExtension = {}
updatedExtension[extensionName] = Object.assign({}, currentData, action.data)

return Object.assign({}, state, updatedExtension)
```